crates/primitives/src/transaction/mod.rs

```diff
@@ -871,7 +872,34 @@ impl TransactionSigned {
         if num_txes < *PARALLEL_SENDER_RECOVERY_THRESHOLD {
             txes.into_iter().map(|tx| tx.recover_signer()).collect()
         } else {
-            txes.into_par_iter().map(|tx| tx.recover_signer()).collect()
+            let some_transaction_invalid = Arc::new(Mutex::new(false));
+            let recovered_signers: Arc<Mutex<Vec<Address>>>= Arc::new(Mutex::new(Vec::new()));
+
+            let chunks = rayon::current_num_threads();
+            let chunk_size: usize = num_txes / chunks;
+
+            let mut iter = txes.into_iter();
+
+            let vectors: Vec<Vec<&TransactionSigned>> = (0..chunks)
+            .map(|_| iter.by_ref().take(chunk_size).collect())
+            .collect();
+
+            vectors.into_par_iter().for_each
+            (|chunk_txes| {
+                for tx in chunk_txes {
+                    if let Some(sender) = tx.recover_signer(){
+                        recovered_signers.lock().unwrap().push(sender);
```

Comment:
can we use a channel here instead like we do in some of the stages to avoid having a contentious mutex? this would also let us send results over the channel(s) instead of having a mutex to verify that no tx was invalid

Reply:
Sure, I will work on that. However, I don't know how to compare the performance. Can you suggest me how to do it?